django/db/models/base.py

```diff
@@ -261,7 +262,10 @@ def __new__(cls, name, bases, attrs):
                         'with field of similar name from '
                         'abstract base class %r' % (field.name, name, base.__name__)
                     )
-                new_class.add_to_class(field.name, copy.deepcopy(field))
```

Comment:
Virtual fields are badly mislabeled. Actually any field can be virtual (as in it doesn't have a column), but what virtual_fields does is that when a subclass is created virtual fields are always copied to the new class, and contribute_to_class is called. This has an effect where if you have a virtual field in parent class, then in parent class that field's model is set to parent class, but in child class the field's model is set to the child class. For other fields the fields are not copied, and thus the model is always set to the originating model. So, virtual fields are actually only about copying and setting the model. The reason the model is needed is get_content_type() in GenericRelation - it needs to get different content type for child as compared to the parent model. This could likely be fixed, but this is definitely out of scope for 1.8.